### PR TITLE
Enable producing native debug symbols in canary/dev releases

### PR DESCRIPTION
**Pull request by marcins**
# ↪️ Pull Request

Added `debug = true` to the canary release profile, this produces binaries with symbol debug information included. Existing `strip` operations take care of stripping these back out for the `@parcel/rust` package itself, however before we `strip` we now use platform specific tools to extract debug symbols out to a separate file / artifact.

At "build and release" time these are combined into a single `debug-symbols` artifact for ease of use. Any additional work is gated to only the dev/canary releases - there's a minor impact to build time, especially uploading the combined artifact, but it's pretty minor.

This has no impact on final `@parcel/rust` binary size (as the debug info is stripped), and it will be on our end when we bump to a canary to download  the debug symbols artifact and upload it to Sentry.

Example dev-release with these changes: https://github.com/parcel-bundler/parcel/actions/runs/8902598798

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
             os: macos-latest
             target: aarch64-apple-darwin
 
-          - name: macos-latest
+          - name: x86_64-apple-darwin
```
Comment (marcins) [inline .github/workflows/release.yml:38]: We talked about this briefly in sync this morning.. maybe it doesn't belong in this PR, but hopefully it doesn't conflict.

**Review by mattcompiles — APPROVED** [review]

**Review by yamadapc — APPROVED** [review]

**Comment by marcins** [comment]
Just leaving some notes here from some "offline" conversations:
* Github has a default 90 day retention for artifacts - if the size of the new artifacts proves problematic, we have a few options: 
 * remove the "convenience" single symbols artifact, just use the separate ones (which will complicate our consumption side slightly)
 * change the retention for the separate artifacts to 1 day, or work out if we can delete them after the combined artifact is produced
 * look into whether we can publish these directly or elsewhere

**Comment by yamadapc** [comment]
I'm removing this from the merge queue because https://github.com/parcel-bundler/parcel/pull/9658 needs to be merged first.
